File: MachineCoding/ParkingLot/ParkingLot/src/main/java/org/example/PaymentContext.java
package org.example;

public class PaymentContext {
    private PaymentStrategy paymentStrategy;

    // Set the strategy dynamically
    public void setPaymentStrategy(PaymentStrategy paymentStrategy) {
        this.paymentStrategy = paymentStrategy;
    }

    // Execute payment using the chosen strategy
    public void executePayment(double amount) {
        if (paymentStrategy == null) {
            throw new IllegalStateException("Payment strategy not set!");
        }
        paymentStrategy.pay(amount);
    }
}

File: devpractice/machinecoding/ParkingLot/ParkingLot/PaymentContext.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ParkingLot
{
    public class PaymentContext
    {
        public void payment(paymentStrategy strategy, double amount)
        {
            strategy.pay(amount);
        }
    }
}
